packages/github/src/interfaces.ts

```diff
@@ -1,39 +1,29 @@
-/* eslint-disable @typescript-eslint/no-explicit-any */
+// TODO: https://github.com/actions/toolkit/issues/291, ESLint chokes on the | syntax
+/* eslint-disable no-undef */
+import Webhooks from '@octokit/webhooks'
+type WebhookPayload =
+  | Webhooks.WebhookPayloadPush
+  | Webhooks.WebhookPayloadPullRequest
+  | Webhooks.WebhookPayloadPullRequestReview
+  | Webhooks.WebhookPayloadPullRequestReviewComment
+  | Webhooks.WebhookPayloadStatus
+  | Webhooks.WebhookPayloadIssues
+  | Webhooks.WebhookPayloadIssueComment
+  | Webhooks.WebhookPayloadRelease
+  | Webhooks.WebhookPayloadRepositoryDispatch
+  | Webhooks.WebhookPayloadCheckRun
+  | Webhooks.WebhookPayloadDeployment
+  | Webhooks.WebhookPayloadCheckSuite
+  | Webhooks.WebhookPayloadWatch
+  | Webhooks.WebhookPayloadDeploymentStatus
+  | Webhooks.WebhookPayloadCreate
+  | Webhooks.WebhookPayloadDelete
+  | Webhooks.WebhookPayloadProjectCard
+  | Webhooks.WebhookPayloadPageBuild
+  | Webhooks.WebhookPayloadFork
+  | Webhooks.WebhookPayloadGollum
+  | Webhooks.WebhookPayloadMilestone
+  | Webhooks.WebhookPayloadProject
+  | Webhooks.WebhookPayloadLabel
 
-export interface PayloadRepository {
```

Comment:
if we re-export PayloadRepository then can we avoid a new major version?

Reply:
I don't believe that there's a great way to keep back-compat and set us on a good foot going forward. 

We took some of the events and combined them into one big type where we marked each field as optional in `WebhookPayload` This doesn't really set us on a great path forward because it makes it hard to tell which fields should exist on the event.

For example, a push event would not have an `issue` or a `pull_request`, but they are marked as optional when consuming. Ideally, this would be distinct from fields that are actually optional fields, but there is no way to tell now.

It conveys the wrong message to a user that those fields are optional, when in fact they depend on the type of event. Our users who want type safety should be casting based on the type of event, we shouldn't combine every event into one big interface with each field is marked as optional, it makes it harder to consume.

It also sends the wrong message because the individual fields depend on the type of event. For example the "pull_request" field on a PullRequestReview is a different object with different fields from the `pull_request` field on a "PullRequest" event.  `payload.pull_request.commits` is only a valid field for PullRequests but not PullRequestReviews. 

By setting these as union types, we enforce that users need to do a type assertion using the `as` keyword, or ignore types by using these as `any` objects